Model/Product/ReplicaManager.php

```diff
@@ -169,55 +194,79 @@ protected function getPossibleMagentoReplicaSettings(string $indexName, int $sto
     /**
      * @inheritDoc
      */
-    public function handleReplicas(string $indexName, int $storeId, array $primaryIndexSettings): void
+    public function handleReplicas(string $primaryIndexName, int $storeId, array $primaryIndexSettings): void
     {
         // TODO: Determine if InstantSearch is a hard requirement (i.e. headless implementations may still need replicas)
         if ($this->configHelper->isInstantEnabled($storeId)
-            && $this->hasReplicaConfigurationChanged($indexName, $storeId))
-        {
-            // TODO: Handle ranking adjustments when toggling virtual vs standard replicas
-            $addedReplicas = $this->setReplicasOnPrimaryIndex($indexName, $storeId);
-            $this->configureRanking($indexName, $storeId, $addedReplicas, $primaryIndexSettings);
+            && $this->hasReplicaConfigurationChanged($primaryIndexName, $storeId)) {
+            $addedReplicas = $this->setReplicasOnPrimaryIndex($primaryIndexName, $storeId);
+            $this->configureRanking($primaryIndexName, $storeId, $addedReplicas, $primaryIndexSettings);
         }
     }
 
     /**
-     * @param $indexName
+     * @param $primaryIndexName
      * @param int $storeId
-     * @return string[] Replicas added by this operation
+     * @return string[] Replicas added or modified by this operation
      * @throws LocalizedException
      * @throws NoSuchEntityException
      * @throws AlgoliaException
      */
     protected function setReplicasOnPrimaryIndex($indexName, int $storeId): array
     {
         $sortingIndices = $this->configHelper->getSortingIndices($indexName, $storeId);
-        $newMagentoReplicas = $this->transformSortingIndicesToReplicaSetting($sortingIndices);
-        $oldMagentoReplicas = $this->getMagentoReplicaConfigurationFromAlgolia($indexName, $storeId);
-        $nonMagentoReplicas = $this->getNonMagentoReplicaConfigurationFromAlgolia($indexName, $storeId);
-        $oldMagentoReplicaIndices = $this->getBareIndexNamesFromReplicaSetting($oldMagentoReplicas);
-        $newMagentoReplicaIndices = $this->getBareIndexNamesFromReplicaSetting($newMagentoReplicas);
+        $newMagentoReplicasSetting = $this->transformSortingIndicesToReplicaSetting($sortingIndices);
+        $oldMagentoReplicasSetting = $this->getMagentoReplicaConfigurationFromAlgolia($indexName, $storeId);
+        $nonMagentoReplicasSetting = $this->getNonMagentoReplicaConfigurationFromAlgolia($indexName, $storeId);
```

Comment:
It's not clear to me why we need to fetch replicas not belonging to Magento ? Shouldn't we assume that Magento is the only source of truth when it comes to apply settings to its own indices ?

Reply:
Great question. Unfortunately that is not the case anymore as tools like Merchandising Studio can create sorting strategies independent of Magento that the integration could potentially blow away destroying merchandising work in the process.

Reply:
Oh I see, and those MS indices have the exact same prefix I guess ?

Reply:
I'm not sure if it's documented but from my testing I've observed that the MS sorting strategies add their own prefix like `cat_sorting_` + the primary index + attribute name. e.g.

`cat_sorting_magento2_dev_default_products_activity` 

As long as this convention is preserved we can rely on our own prefix match. Some further examples:

<img width="875" alt="2024-06-12_09-54-12" src="https://github.com/algolia/algoliasearch-magento-2/assets/986313/03fbccf8-e6ef-4455-9f0f-fc2146cd1c13">

The last two indices above are not managed by Magento. The first is an arbitrary replica and the second is created by an MS "sorting strategy". The new approach will preserve both of these.